docs/Datadog.Trace/README.md

```diff
@@ -0,0 +1,220 @@
+* Setting `TracerSettings.AnalyticsEnabled` has similarly been marked obsolete.
+* Environment variable `DD_TRACE_LOG_PATH` is deprecated. Use `DD_TRACE_LOG_DIRECTORY` instead.
+
+### Introduction of ISpan and IScope
+
+In version 2.0 the public `Tracer` API has been updated to expose `IScope` and `ISpan` instead of `Scope` and `Span`. In most cases, this should not require any changes to your code, but if you are currently using explicit types (instead of inferred types), then you will need to replace usages of `Scope` with `IScope` and `Span` with `ISpan`:
+
+```csharp
+using Datadog.Trace;
+
+// no changes required here
+using (var scope = Tracer.Instance.StartActive("my-operation")
+{
+}
+
+// No Longer compiles (incorrect usage of Scope)
+using (Scope scope = Tracer.Instance.StartActive("my-operation")
+{
+}
+
+// Correct usage with explicit type
+using (IScope scope = Tracer.Instance.StartActive("my-operation")
+{
+}
+```
+
+### Incorrect integration names are ignored
+
+In version 2.0, any changes made to an `IntegrationSettings` object for an unknown `IntegrationId` will not be persisted. Instead, a warning will be logged describing the invalid access.
+
+```csharp
+using Datadog.Trace;
+
+var tracerSettings = TracerSettings.FromDefaultSources();
+
+// Accessing settings for an unknown integration will log a warning
+var settings = tracerSettings .Integrations["MyRandomIntegration"];
+
+// changes are not persisted
+settings.Enabled = false; 
+
+// isEnabled is null, not false
+bool? isEnabled = tracerSettings.Integrations["MyRandomIntegration"].Enabled;
+```
+
+### Default value of `DD_TRACE_ROUTE_TEMPLATE_RESOURCE_NAMES_ENABLED` is now `true`
+
+Version 1.26.0 added support for the `DD_TRACE_ROUTE_TEMPLATE_RESOURCE_NAMES_ENABLED` feature flag, which enabled improved span names for ASP.NET and ASP.NET Core automatic instrumentation spans, an additional span for ASP.NET Core requests, and additional tags. 
+
+In version 2.0, `DD_TRACE_ROUTE_TEMPLATE_RESOURCE_NAMES_ENABLED` is **enabled** by default. Due to the change in span names, you may need to update your monitors and dashboards to use the new resource names. 
+
+If you do not wish to take advantage of the improved route names, you can disabled the feature by setting the `DD_TRACE_ROUTE_TEMPLATE_RESOURCE_NAMES_ENABLED` environment variable to `0`.
+
+### CallSite instrumentation has been removed
+
+From version 2.0, callsite automatic instrumentation has been removed and replaced with calltarget instrumentation. This has been the default mode [since version 1.28.0](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.28.0). Calltarget provides performance improvements over callsite, but we no longer support instrumenting some custom implementations of `DbCommand`. If you find spans are missing from your traces after upgrading, please raise an issue on GitHub, or contact [support](https://docs.datadoghq.com/help).
```

Comment:
It's fine to use "CallSite" and "callsite" internally, but all public docs should use "call-site instrumentation."

```suggestion
### Call-site instrumentation has been removed

From version 2.0, call-site automatic instrumentation has been removed and replaced with call-target instrumentation. This has been the default mode [since version 1.28.0](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.28.0). Call-target instrumentation provides performance improvements over call-site instrumentation, but we no longer support instrumenting some custom implementations of `DbCommand`. If you find spans are missing from your traces after upgrading, please raise an issue on GitHub, or contact [support](https://docs.datadoghq.com/help).
```